From aea46fe3d78c67e4cbada3ebc37b0cdae7d77107 Mon Sep 17 00:00:00 2001
From: Guneshwor Singh <guneshwor.o.singh@intel.com>
Date: Mon, 24 Oct 2016 16:23:20 +0530
Subject: [PATCH 0530/1132] Register masters only if RT700 is selected in
 config

This info should come from BIOS settings. Right now it is
not available, so using WA to avoid registering sdw masters
if used in I2S mode.

Change-Id: Id5cad0a8e53507481b65a56e942b4eedc8f0a108
Signed-off-by: Guneshwor Singh <guneshwor.o.singh@intel.com>
---
 sound/soc/intel/skylake/cnl-sst.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/intel/skylake/cnl-sst.c b/sound/soc/intel/skylake/cnl-sst.c
index a124dab..a81e1b0 100644
--- a/sound/soc/intel/skylake/cnl-sst.c
+++ b/sound/soc/intel/skylake/cnl-sst.c
@@ -509,6 +509,7 @@ static int cnl_ipc_init(struct device *dev, struct skl_sst *cnl)
 	return 0;
 }
 
+#if IS_ENABLED(CONFIG_SND_SOC_RT700)
 static int skl_register_sdw_masters(struct device *dev, struct skl_sst *dsp,
 			void __iomem *mmio_base, int irq)
 {
@@ -655,6 +656,7 @@ static int skl_register_sdw_masters(struct device *dev, struct skl_sst *dsp,
 	cnl_sdw_int_enable(dsp->dsp, 1);
 	return 0;
 }
+#endif
 
 static void skl_unregister_sdw_masters(struct skl_sst *ctx)
 {
@@ -723,11 +725,13 @@ int cnl_sst_dsp_init(struct device *dev, void __iomem *mmio_base, int irq,
 		return ret;
 	}
 
+#if IS_ENABLED(CONFIG_SND_SOC_RT700)
 	ret = skl_register_sdw_masters(dev, cnl, mmio_base, irq);
 	if (ret) {
 		dev_err(cnl->dev, "%s SoundWire masters registration failed\n", __func__);
 		return ret;
 	}
+#endif
 
 	return 0;
 }
-- 
2.7.4

